model/godleyIcon.h

```diff
@@ -112,6 +112,12 @@ namespace minsky
     GodleyTable table;
     /// updates the variable lists with the Godley table
     void update();
+    
+    GodleyIcon* clone() const override {
+      auto r=new GodleyIcon(*this);
+      r->update();
+      return r;
```

Comment:
The problem with this approach is that it leaves the new cloned GodleyIcon purportedly a member of the original group, but the controller items of the contained variables will be null, because the cloned GodleyIcon is not actually a member of the group.

Because of this, it may prove better, or more consistent to create a dummy group for the cloned Godley icon to live in while the update takes place.

Reply:
Thank you, that was a very useful suggestion. What I've done seems to work, but also looks a little hacky to me, i.e., I feel there may be a better way to write the bit of code I've added...